src/NumericalAlgorithms/Spectral/Spectral.hpp

```diff
@@ -64,6 +76,11 @@ std::ostream& operator<<(std::ostream& os, const Basis& basis) noexcept;
  * exact to polynomial order \f$p=2N-1\f$. Gauss-Lobatto quadrature is exact
  * only to polynomial order \f$p=2N-3\f$, but includes collocation points at the
  * domain boundary.
+ *
+ * \warning `CellCentered` are intended to be used with the `FiniteDifference`
```

Comment:
The use of plural here ("are intended", and later "they could be", "do not implement") is surprising... but I see it's because you'll be adding `FaceCentered` in the next commit, so I'll allow it ;)